sktime/forecasting/moirai_forecaster.py

```diff
@@ -0,0 +1,461 @@
+    >>> X = pd.DataFrame(X, columns=["x1", "x2"], index=index)
+    >>> df = pd.concat([y, X], axis=1)
+    >>> morai_forecaster.fit(y, X=X)
+    >>> X_test = pd.DataFrame(np.random.normal(0, 1, (10, 2)),
+    ...                      columns=["x1", "x2"],
+    ...                      index=pd.date_range("2020-01-31", periods=10, freq="D"),
+    ... )
+    >>> forecast = morai_forecaster.predict(fh=range(1, 11), X=X_test)
+
+    References
+    ----------
+    .. [1] https://huggingface.co/collections/sktime/moirai-variations-66ba3bc9f1dfeeafaed3b974
+    .. [2] https://pypi.org/project/uni2ts/1.1.0/
+    """
+
+    _tags = {
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:pred_int": False,
+        "python_dependencies": [
+            "gluonts",
+            "torch",
+            "jax",
+            "jaxtyping",
+            "einops",
+            "huggingface-hub",
+            "hydra-core",
+            "lightning",
+        ],
+        "X_inner_mtype": ["pd.DataFrame", "pd-multiindex"],
+        "y_inner_mtype": ["pd.DataFrame", "pd-multiindex"],
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "capability:global_forecasting": True,
+        "authors": ["gorold", "chenghaoliu89", "liu-jc", "benheid", "pranavvp16"],
+        # gorold, chenghaoliu89, liu-jc are from SalesforceAIResearch/uni2ts
+        "maintainers": ["pranavvp16"],
+        "python_dependencies_alias": {"salesforce-uni2ts": "uni2ts"},
+    }
+
+    def __init__(
+        self,
+        checkpoint_path: str,
+        context_length=200,
+        patch_size=32,
+        num_samples=100,
+        num_feat_dynamic_real=0,
+        num_past_feat_dynamic_real=0,
+        map_location=None,
+        broadcasting=False,
+        deterministic=False,
+        batch_size=32,
+        use_source_package=False,
+    ):
+        super().__init__()
+        self.checkpoint_path = checkpoint_path
+        self.context_length = context_length
+        self.patch_size = patch_size
+        self.num_samples = num_samples
+        self.num_feat_dynamic_real = num_feat_dynamic_real
+        self.num_past_feat_dynamic_real = num_past_feat_dynamic_real
+        self.map_location = map_location
+        self.broadcasting = broadcasting
+        self.deterministic = deterministic
+        self.batch_size = batch_size
+        self.use_source_package = use_source_package
+
+        if self.broadcasting:
+            self.set_tags(
+                **{
+                    "y_inner_mtype": "pd.DataFrame",
+                    "X_inner_mtype": "pd.DataFrame",
+                    "capability:global_forecasting": False,
+                }
+            )
+
+    # Apply a patch for redirecting imports to sktime.libs.uni2ts
+    @patch.dict("sys.modules", {"uni2ts": sktime.libs.uni2ts})
```

Comment:
Why is patching necessary? Could you just adapt the imports in the vendored library?